mantra/representations/simplicial_connectivity.py

```diff
@@ -0,0 +1,300 @@
+                elif "adjacency" in self.connectivity_name:
+                    data[connectivity_name] = torch.zeros(
+                        [shape[rank_idx], shape[rank_idx]],
+                        layout=torch.sparse_coo,
+                    ).coalesce()
+
+        return data
+
+
+class IncidenceSCTransform(ConnectivitySCTransform):
+    """Add incidences of a simplicial complex."""
+
+    def __init__(self, signed: bool, index=False):
+        super().__init__(signed, "incidence", index)
+
+    def generate_matrix(
+        self, simplex_trie: SimplexTrie, rank: int, max_rank: int
+    ) -> scipy.sparse.csr_matrix:
+        idx_simplices, idx_faces, values = [], [], []
+
+        simplex_dict_d = {
+            tuple(sorted(simplex)): i
+            for i, simplex in enumerate(self.skeleton(simplex_trie, rank))
+        }
+        simplex_dict_d_minus_1 = {
+            tuple(sorted(simplex)): i
+            for i, simplex in enumerate(self.skeleton(simplex_trie, rank - 1))
+        }
+        for simplex, idx_simplex in simplex_dict_d.items():
+            for i, left_out in enumerate(np.sort(list(simplex))):
+                idx_simplices.append(idx_simplex)
+                values.append((-1) ** i)
+                face = frozenset(simplex).difference({left_out})
+                idx_faces.append(simplex_dict_d_minus_1[tuple(sorted(face))])
+
+        if len(values) != (rank + 1) * len(simplex_dict_d):
+            raise RuntimeError("Error in computing the incidence matrix.")
+
+        boundary = csr_matrix(
+            (values, (idx_faces, idx_simplices)),
+            dtype=np.float32,
+            shape=(
+                len(simplex_dict_d_minus_1),
+                len(simplex_dict_d),
+            ),
+        )
+
+        if not self.signed:
+            boundary = abs(boundary)
+        if self.index:
+            return simplex_dict_d_minus_1, simplex_dict_d, boundary
+        return boundary
+
+
+class UpLaplacianSCTransform(ConnectivitySCTransform):
+    """Add Up Laplacian of a simplicial complex."""
+
+    def __init__(self, signed: bool, index: bool = False):
+        super().__init__(signed, "up_laplacian", index=index)
+
+    def generate_matrix(
+        self, simplex_trie: SimplexTrie, rank: int, max_rank: int
+    ):
+        incidence_matrix_transform = IncidenceSCTransform(
+            self.signed, index=True
+        )
+
+        if rank < max_rank and rank >= 0:
+            row, _, B_next = incidence_matrix_transform.generate_matrix(
+                simplex_trie, rank + 1, max_rank
+            )
+            L_up = B_next @ B_next.transpose()
+        else:
+            raise ValueError(
+                f"Rank should larger than 0 and <= {max_rank - 1} (maximal dimension cells-1), got {rank}"
+            )
+        if not self.signed:
+            L_up = abs(L_up)
+        if self.index:
+            return row, L_up
+
+        return L_up
+
+
+class DownLaplacianSCTransform(ConnectivitySCTransform):
```

Comment:
I believe this would also benefit from being more "verbose" as a name (minus the "Transform" part).